contracts/XVSVault/XVSVault.sol

```diff
@@ -187,6 +190,31 @@ contract XVSVault is XVSVaultStorage, ECDSA {
         emit Deposit(msg.sender, _rewardToken, _pid, _amount);
     }
 
+    /**
+     * @notice Claim rewards for pool
+     * @param _account The account for which to claim rewards
+     * @param _rewardToken The Reward Token Address
+     * @param _pid The Pool Index
+     */
+    function claim(address _account, address _rewardToken, uint256 _pid) external nonReentrant {
+        _ensureValidPool(_rewardToken, _pid);
+        PoolInfo storage pool = poolInfos[_rewardToken][_pid];
+        UserInfo storage user = userInfos[_rewardToken][_pid][_account];
+        _updatePool(_rewardToken, _pid);
+        if (user.amount > 0) {
+            uint256 pending = user.amount.sub(user.pendingWithdrawals).mul(pool.accRewardPerShare).div(1e12).sub(
+                user.rewardDebt
+            );
+
+            user.rewardDebt = user.amount.sub(user.pendingWithdrawals).mul(pool.accRewardPerShare).div(1e12);
+
+            if (pending > 0) {
+                IXVSStore(xvsStore).safeRewardTransfer(_rewardToken, _account, pending);
+                emit Claim(_account, _rewardToken, _pid, pending);
```

Comment:
I agree. For any state changes we should emit events. I have updated the code.

Reply:
Does it make sense to "try" to transfer 0 XVS?

Reply:
It will not cause any side effects. Right now in the XVSVault contract we don't check if the reward is 0 or not when transferring. So I think we should maintain the same standard in the `claim` function also.